VSRAD.Syntax/Core/LexerParserFactory.cs

```diff
@@ -14,32 +15,50 @@ public struct LexerParser
 
     internal partial class DocumentFactory
     {
+        private Dictionary<AsmType, LexerParser> _asmSpecificLexerParser;
+
         private LexerParser? GetLexerParser(AsmType asmType)
         {
-            var instructionManager = _instructionManager.Value;
+            InitializeLexerParser();
+
+            if (_asmSpecificLexerParser.TryGetValue(asmType, out var lexerParser))
```

Comment:
(Subjective) I think that keeping the `switch` as it used to be and just replacing the `new` calls with `.Instance` accessors is a clearer approach that results in less code and less runtime overhead.

Reply:
Thank you, I agree with you.
Fixed in 4dae221